core/src/main/java/nu/yona/server/analysis/entities/IntervalActivity.java

```diff
@@ -0,0 +1,53 @@
+package nu.yona.server.analysis.entities;
+
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.UUID;
+
+import javax.persistence.Entity;
+
+import nu.yona.server.entities.EntityWithID;
+import nu.yona.server.goals.entities.Goal;
+
+@Entity
+public abstract class IntervalActivity extends EntityWithID
+{
+	private UUID goalID;
+	private UUID userAnonymizedID;
+	private ZonedDateTime zonedStartTime;
```

Comment:
I am not sure if we want all date/times zoned. For a date on goal creation/message creation I think a UTC date is appropriate, it can be converted to the user zoned date when sending it to the client (or by the app itself). For the `DayActivity` and `WeekActivity` the time zone really does matter, it is impossible to save these as UTC, because around the globe the days (and weeks) start later or sooner. So I would say to keep all dates in UTC `Date` except for the `DayActivity` and `WeekActivity`.